File: go-sdk/pkg/models/operations/putfoldersidnumberremovesharedlink.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PutFoldersIDNumberRemoveSharedLinkPathParams struct {
	FolderID string `pathParam:"style=simple,explode=false,name=folder_id"`
}

type PutFoldersIDNumberRemoveSharedLinkQueryParams struct {
	Fields string `queryParam:"style=form,explode=true,name=fields"`
}

type PutFoldersIDNumberRemoveSharedLinkRequestBody struct {
	SharedLink map[string]interface{} `json:"shared_link,omitempty"`
}

type PutFoldersIDNumberRemoveSharedLinkRequest struct {
	Retries     *utils.RetryConfig
	PathParams  PutFoldersIDNumberRemoveSharedLinkPathParams
	QueryParams PutFoldersIDNumberRemoveSharedLinkQueryParams
	Request     *PutFoldersIDNumberRemoveSharedLinkRequestBody `request:"mediaType=application/json"`
}

type PutFoldersIDNumberRemoveSharedLinkResponse struct {
	ClientError *shared.ClientError
	ContentType string
	Folder      *shared.Folder
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/putfoldersidnumberremovesharedlink.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PutFoldersIdNumberRemoveSharedLinkPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=folder_id" })
  folderId: string;
}


export class PutFoldersIdNumberRemoveSharedLinkQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields: string;
}


export class PutFoldersIdNumberRemoveSharedLinkRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=shared_link" })
  sharedLink?: Record<string, any>;
}


export class PutFoldersIdNumberRemoveSharedLinkRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PutFoldersIdNumberRemoveSharedLinkPathParams;

  @SpeakeasyMetadata()
  queryParams: PutFoldersIdNumberRemoveSharedLinkQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PutFoldersIdNumberRemoveSharedLinkRequestBody;
}


export class PutFoldersIdNumberRemoveSharedLinkResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  folder?: shared.Folder;

  @SpeakeasyMetadata()
  statusCode: number;
}
